Remove seperation between ArcGis Feature Service and ArcGIS Map Service items in browser

nyalldawson opened the pull request:
Builds off https://github.com/qgis/QGIS/pull/40697:

Replaces the separate "ArcGis Feature Service" and "ArcGIS Map Service" browser items/connections with a single, unified "ArcGIS Rest Servers" connection and browser node which shows a definitive view of the server exposing both service types.

This makes it MUCH easier to manage and browse these servers, and matches user instead of the awkward forced separation between the layer types.

nyalldawson committed:
[needs-docs] Remove seperation between ArcGis Feature Service
and ArcGIS Map Service items in browser

and replace them with a single, unified "ArcGIS Rest Servers" item
which shows a definitive view of the server exposing both
service types.

nirvn commented:
I'm a big +1 on this move. Makes a lot of sense and de-clutters our browser panel as well as data source dialog.

nyalldawson committed:
Replace ArcGIS REST data source manager list with a view based on the browser model

This has MANY benefits:
- Shows the merged feature service/map service view we use in browser
- Avoids the HUGE main thread ui hang when connecting to servers with
many layers (could be 20 mins+ for some services)
- Avoids bombing out and showing incomplete lists when a server reports
an error when connecting to any of the services it advertises (happens
often)

nyalldawson committed:
Adding works again

nyalldawson committed:
Remove useless projection selector -- arcgis rest layers are only available in a single crs

nyalldawson committed:
Handle image formats

nyalldawson committed:
Remove redundant AMS gui components, they are all handled by arcgis REST unified classes

nyalldawson committed:
Remove non-functional querying code from feature server source select

nyalldawson committed:
Auto migrate mapserver connections to ArcGIS REST connections

nyalldawson committed:
QStringLiteral

nyalldawson commented:
I've also added other improvements here:

1. There's now a unified "ArcGIS REST Server" tab in the Data Source Manager, which shows the combined feature service/map service views of servers
2. The Data Source Manager widget is now based off browser models, so it's nice and responsive and loads in background threads, and only loads data on demand. Previously we walked the ENTIRE server in one blocking call on the main thread, which would lead to 10 min+ ui hangs on large servers. Ouch!
3. Removed the never-hooked-up, non functional query buttons for feature service layers
4. Made the search line edit actually work

nyalldawson committed:
Use correct dialog in source select

nirvn commented on src/providers/arcgisrest/qgsarcgisrestdataitemguiprovider.cpp:
```diff
@@ -142,21 +160,21 @@ void QgsAfsDataItemGuiProvider::deleteConnection( QgsDataItem *item )
 }
 
-void QgsAfsDataItemGuiProvider::loadConnections( QgsDataItem *item )
+void QgsArcGisRestDataItemGuiProvider::loadConnections( QgsDataItem *item )
```
Do we need to add code to handle pre 3.18 saved connections that'd include ArcgisMapServer?

nyalldawson replied:
That's done here: https://github.com/qgis/QGIS/pull/40700/files#diff-5fab4f0ef4a0231dcfa1488222c365a5955f1df79cfaf06e4086275ea6bd542bR611

nirvn commented on src/providers/arcgisrest/qgsarcgisrestdataitems.cpp:
```diff
@@ -0,0 +1,674 @@
+
+
+//
+// QgsAfsGroupsItem
```
There seems to be a bunch of these section comments that are out of sync with the subsequent class name, e.g. QgsAfsGroupsItem vs QgsArcGisPortalGroupsItem.

nyalldawson replied:
Fixed, thanks

nyalldawson committed:
Update comments
